### DISCUSSION TITLE
Is it possible to have multiple TF configurations planned/applied from a single PR?

### DISCUSSION THREAD

**Discussion by utku-ki** [dflook/terraform-github-actions#161] [Q&A] [answered]
I have a repository where I have multiple Terraform configurations in different directories.

In a single PR, I might change more than one of them.

Would the `terraform plan` comment on the PR work correctly in this case, so that when the PR is merged, `terraform apply` would pick the right plan from the merged PR and apply it for the configuration it is running for?

Thanks for the answer and for the great project.

**Comment by dflook** [comment, marked as answer]
Hi @utku-ki , you can have multiple the plans/applies for the same PR. 

The PR comment is identified using the values of the action inputs. The input values for the plan action must be the same in the corresponding apply action. Since the `path` input will be different for each of your plans, it will find the correct comment.

**Reply by utkuozdemir** [reply]
Thanks for the clarification. In fact, I also set up a small project to try it in the meantime and it worked as I expected.

Cheers.

**Reply by rdhar** [reply]
Hey @utkuozdemir, taking inspiration from this discussion exactly a year later, I managed to cobble together a workflow in [rdhar/aws-terraform-multiple-environments](https://github.com/rdhar/aws-terraform-multiple-environments) which executes terraform plan/apply across multiple environments from just one PR. 

Opened a separate [discussion](https://github.com/dflook/terraform-github-actions/discussions/241); let me know of your thoughts or feedback on this! Thanks, @dflook, for your continued time and effort on this project.
